theia-tree-editor/src/browser/interfaces.ts

```diff
@@ -91,7 +92,7 @@ export namespace TreeEditor {
          * @param node The tree node
          * @returns The ui schema for the node's data or undefined to generate a ui schema.
          */
-        getUiSchemaForNode(node: Node): UISchemaElement | undefined;
+        getUiSchemaForNode(node: Node): MaybePromise<UISchemaElement> | undefined;
```

Comment:
I'd change the return type to `MaybePromise<UISchemaElement | undefined>` because the Promise can also evaluate to `undefined`.